src/tests/daos_perf.c

```diff
@@ -1398,10 +1403,25 @@ main(int argc, char **argv)
 	}
 
 	if (ts_mode == TS_MODE_VOS) {
+		if (ts_ctx.tsc_mpi_size > 1 &&
+		    (access("/etc/daos_nvme.conf", F_OK) != -1)) {
+			fprintf(stderr,
+				"no support: multi-proc vos_perf with NVMe\n");
+			return -1;
+		}
 		ts_ctx.tsc_cred_nr = -1; /* VOS can only support sync mode */
-		if (strlen(ts_pmem_file) == 0)
-			strcpy(ts_pmem_file, "/mnt/daos/vos_perf.pmem");
-
+		if (strlen(ts_pmem_file) == 0) {
+			int len = snprintf(NULL, 0, "/mnt/daos/vos_perf%d.pmem",
+					   ts_ctx.tsc_mpi_rank) + 1;
+			snprintf(ts_pmem_file, len, "/mnt/daos/vos_perf%d.pmem",
```

Comment:
```suggestion
			snprintf(ts_pmem_file, sizeof(ts_pmem_file), "/mnt/daos/vos_perf%d.pmem",
```

Reply:
The point of len is to get the string length to ensure we do sprintf for the appropriate buffer size. Sizeof is not accurate as it can technically be more than actual string length.

Reply:
My point here is just why we have to calculate the len if we can specify just the size of buffer "ts_pmem_file"?
Passing size into snprintf we should just prevent write out of the buffer.